6_nodejs/week1/avg.js

```diff
@@ -8,6 +8,30 @@ function calculateAverage(numbers) {
 }
 
 //retrieves the command line arguments
-const args = process.argv.slice(2);
-const average = calculateAverage(args);
+const arguments = process.argv.slice(2);
+const average = calculateAverage(arguments);
 console.log(average);
+
+//--------Optional question--------------------------------------
+//arguments are mix of numbers and string
+function printNumbersAndStrings(args) {
+  if (args.length === 0) {
+    console.log("There is no arguments.");
+    return;
+  }
+  const numbers = [];
+  const strings = [];
+  for (let i = 0; i < args.length; i++) {
+    const num = parseFloat(args[i]);
+    if (!isNaN(num)) {
+      numbers.push(num);
+    } else {
+      strings.push(args[i]);
+    }
+  }
+  console.log("Numbers:", numbers);
+  console.log("Strings:", strings);
```

Comment:
This is one way to handle when the user accidentally provides strings and numbers. However, as the purpose of the code is to calculate the average of **numbers**, we shall treat inputs that contain **strings** as **incorrect inputs**. Thus, if the user inputs a string as an argument, we should handle it as an **error** and let the user know their mistake.